test/jdk/javax/net/ssl/HttpsURLConnection/CriticalSubjectAltName.java

```diff
@@ -27,31 +27,60 @@
 //
 
 /*
- * @test
+ * @test id=tls12
  * @bug 6668231
  * @summary Presence of a critical subjectAltName causes JSSE's SunX509 to
  *          fail trusted checks
- * @run main/othervm CriticalSubjectAltName
+ * @library /test/lib
+ * @modules java.base/sun.security.x509 java.base/sun.security.util
+ * @run main/othervm CriticalSubjectAltName TLSv1.2 MD5withRSA
```

Comment:
as far as I could tell, this test doesn't verify any functionality that would require a specific key type, it's simply using MD5 because that was the popular choice in 2008. Can we make it use whatever key type is the default?